mmv1/products/firebaseappcheck/DebugToken.yaml

```diff
@@ -0,0 +1,93 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+!ruby/object:Api::Resource
+name: "DebugToken"
+base_url: projects/{{project}}/apps/{{app_id}}/debugTokens
+self_link: projects/{{project}}/apps/{{app_id}}/debugTokens/{{debug_token_id}}
+update_verb: :PATCH
+update_mask: true
+description: |
+  A debug token is a secret used during the development or integration testing of
+  an app. It essentially allows the development or integration testing to bypass
+  app attestation while still allowing App Check to enforce protection on supported
+  production Firebase services.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    "Official Documentation": "https://firebase.google.com/docs/app-check"
+  api: "https://firebase.google.com/docs/reference/appcheck/rest/v1/projects.apps.debugTokens"
+import_format:
+  [
+    "projects/{{project}}/apps/{{app_id}}/debugTokens/{{debug_token_id}}",
+    "{{project}}/{{app_id}}/{{debug_token_id}}",
+    "{{app_id}}/{{debug_token_id}}",
+  ]
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firebase_app_check_debug_token_basic"
+    # Need the time_sleep resource
+    pull_external: true
+    primary_resource_id: "default"
+    vars:
+      display_name: "Debug Token"
+      token: "00000000-AAAA-BBBB-CCCC-000000000000"
+    test_vars_overrides:
+      # Don't add prefix. This string is a random UUID4
+      token: "\"5E728315-E121-467F-BCA1-1FE71130BB98\""
+    test_env_vars:
+      project_id: :PROJECT_NAME
+parameters:
+  - !ruby/object:Api::Type::String
+    name: app_id
+    description: |
+      The ID of a
+      [Web App](https://firebase.google.com/docs/reference/firebase-management/rest/v1beta1/projects.webApps#WebApp.FIELDS.app_id),
+      [Apple App](https://firebase.google.com/docs/reference/firebase-management/rest/v1beta1/projects.iosApps#IosApp.FIELDS.app_id),
+      or [Android App](https://firebase.google.com/docs/reference/firebase-management/rest/v1beta1/projects.androidApps#AndroidApp.FIELDS.app_id)
+    required: true
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: name
+    description: |
+      The fully-qualified resource name of the debug token.
+    output: true
+    pattern: projects/{{project}}/apps/{{app_id}}/debugTokens/{{debug_token_id}}
+    custom_flatten: 'templates/terraform/custom_flatten/firebase_app_check_debug_token.go.erb'
```

Comment:
Shouldn't custom flatten be put on debug_token_id? Otherwise name doesn't get set.

Reply:
The [response](https://firebase.google.com/docs/reference/appcheck/rest/v1/projects.apps.debugTokens) contains the `name` field but not `debug_token_id`. IIUC, the custom flattener on `name` doesn't alter the value but performs a side effect `d.Set("debug_token_id", ...)` to extract the last segment of `name` as the value.

Reply:
hmm yeah but having a custom flattener like this means you essentially are hacking around `name`. Name effectively becomes input only as a side effect.

Reply:
I think we should remove the custom flattener here.